### PR TITLE
Entities (nodes) identified using CURIEs

### PR DESCRIPTION
**Pull request by cbizon**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,9 +22,7 @@ This README documents the current strawman architecture.  Changes must be made v
     3. ARA sends query messages to KPs
     4. KPs respond to ARAs with Message
 3. All communication between components conforms to the ReasonerAPI Message spec
-4. All nodes and edges in all messages (query and response) are normalized
-    1. SRI will provide tools for normalization
-    2. KPs must use https://nodenormalization-sri.renci.org/apidocs/ and https://edgenormalization-sri.renci.org/apidocs/
+4. Entities in any message (ARS/ARA or ARA/KP) are represented using compact URIs (CURIES), which can be expanded to full IRIs using a biolink-model provided json-ld context file.
```
Comment (patrickkwang) [inline README.md:25]: Does "can be expanded" imply "must be expandable"? i.e. that only CURIE prefixes known to the biolink model can be used?

Comment (cbizon) [reply README.md:25]: Yes, I think that's right.  Probably I should change the language to be clearer.

**Comment by andrewsu** [comment]
So @newgene (Services Provider) and I (Exploring Agent) had the most permissive answer here, that KPs **can** return CURIEs. (Three other teams answered **should**, rest say **must**.)

My rationale for not requiring or even recommending CURIEs is based on the SmartAPI-centric approach of using APIs "in the wild" as KPs. Recall that we create KPs that conform to the BioLink Model using semantic mappings of the inputs and outputs. Take, for example, the API from CTD (documentation at http://ctdbase.org/help/linking.jsp#batchqueries). The SmartAPI record is at https://smart-api.info/ui/0212611d1c670f9107baf00b77f0889a, and the OpenAPI metadata (with semantic mapping) is at https://smart-api.info/api/metadata/0212611d1c670f9107baf00b77f0889a?format=yaml.

Based on that API metadata, consider this CTD API call http://ctdbase.org/tools/batchQuery.go?inputType=chem&inputTerms=D003634&report=genes_curated&format=json, which expects a compound as input (in the form of a MeSH ID) and returns related genes as output.  Here is a snippet of the output:

```
[
   {
      "CasRN": "50-29-3",
      "ChemicalId": "D003634",
      "ChemicalName": "DDT",
      "GeneId": "19",
      "GeneSymbol": "ABCA1",
      "Input": "d003634",
      "Organism": "Homo sapiens",
      "OrganismId": "9606",
      "PubMedIds": "25720399"
   },
   ...
]
```

So in this case, the input parameter `D003634` is not a CURIE -- CTD assumes that you will be supplying a MeSH ID for the `inputTerms` (*).  Similarly in the output, the `GeneId` value is always an NCBI Gene ID. Because the CTD API makes these assumptions consistently, the [SmartAPI semantic mapping file](https://smart-api.info/api/metadata/0212611d1c670f9107baf00b77f0889a?format=yaml) is able to transform the API output into a structure that is BioLink Model compliant.  So because SmartAPI offers a reasonable route to go from API output to BioLink Model in the absence of CURIEs, that is why I think this should be a **can** and not a **must**.

(*) reading the documentation a bit more carefully now, the CTD API will actually accept a CURIE, e.g., http://ctdbase.org/tools/batchQuery.go?inputType=chem&inputTerms=MESH:D003634&report=genes_curated&format=json. But rather than retool this example with a different API, I think it still illustrates the point...

**Comment by cbizon** [comment]
@andrewsu I don't see these as incompatible, though the curie-only version may require a bit more work (or different work) from the services KP.  

Information coming from apis like CTD is going to get turned into curies somewhere to merge with all the other curie data floating around, it's really just a question of where.   I think that in the smartAPI framework, the metadata provided in the registry is sufficient for a general agent like BTE to  handle whatever comes back from a catalogued api, so it's just a question of where that happens.  

If one stood up a server (KP) that proxied calls to the smart api-registered APIs then it could use the BTE library to convert whatevers into curies as needed, right?

**Comment by cbizon** [comment]
> Agreed this should be required. For us it has always has been the case. Although I wonder if some KPs might have concepts for which there is no existing public CURIE? What will they do? Make one up? Mint one?

Great question.   I think that would be a good time to work with the data-modeling group to figure out how to create those curies as needed.    Minting new curies as a stopgap seems perfectly reasonable though.

**Comment by MarkDWilliams** [comment]
I think what Andrew is proposing, and I was initially unclear on this as well, isn't a NOT for requiring CURIES so much as it is proposing an OR for this category. So, the KPs must use CURIES to identify entities OR directly annotate the API with the type of semantic mapping provided. The semantic mapping could be useful for bringing in knowledge sources from outside the consortium. @andrewsu , the sources that are integrated via semantic mapping would still be accessible via CURIE by going through BTE, correct?

**Comment by andrewsu** [comment]
@MarkDWilliams right, I was proposing the "OR" option you stated here

> the KPs must use CURIES to identify entities OR directly annotate the API with the type of semantic mapping provided

And yes as we discussed on the architecture call last week, we will provide a BTE server instance that will return Reasoner Std API with CURIEs (with the understanding that I think that may introduce scalability issues down the line, in which case we would argue for pushing people to directly use the BTE library and/or the smartAPI mapping record).

**Comment by cbizon** [comment]
@ehinderer I suspect that you meant this comment for #11 but since the quesition is here, I'll answer it here :)

I'd say that the ARAs at this point have some options / choices.  They can use ReasonerAPI or they can use things from the SmartAPI registry and call them directly (probably using BTE as a library).  The Proxies in 3a and 3b should allow access to everything via ReasonerAPI, but potentially at the cost of some bottlenecking, so particular ARAs may make different choices about how things should be accessed.

For the tools in 3a/3b, the SmartAPI team is working on a version of 3a, but I could imagine SRI helping on the infrastructure side if needed / desired.   Similarly I imagine SRI handling 3b, but probably making use of some tools that project teams have built or will build (things like data2services or automat).

**Pull request merged**